src/js/hooks/cycleCount/useInventoryValidation.js

```diff
@@ -44,11 +45,18 @@ const useInventoryValidation = ({ tableData }) => {
         cycleCountItems,
         (item) => item?.inventoryItem?.lotNumber,
       );
-      const expirationDates = dataGroupedByLotNumber[row?.inventoryItem?.lotNumber]
-        .map((item) => item?.inventoryItem?.expirationDate);
-      const binLocations = dataGroupedByLotNumber[row?.inventoryItem?.lotNumber]
-        .map((item) => item?.binLocation?.id);
-      if (_.uniq(expirationDates).length !== 1 && _.uniq(binLocations).length !== 1) {
+      // The backend returns expiration date in the "MM/dd/yyyy" format, but we use the
+      // "yyyy-MM-dd'T'hh:mm:ssXXX" format when generating the dates on the frontend. We
+      // convert the dates to a moment for easy comparison.
+      const expirationDates = dataGroupedByLotNumber[row?.inventoryItem?.lotNumber].map((item) => {
+        const expirationDate = item?.inventoryItem?.expirationDate;
+        // TODO: isSame is returning false, even if the dates are the same! I suspect it's trying
+        //       to use local time when parsing "MM/dd/yyyy". Gotta find a way to make this default
+        //       to UTC. This kinda sucks though, maybe we need a more general solution to this...
```

Comment:
would appreciate anyone's help getting this to work cleanly

Reply:
Note that we can't just trim off the time portion of the date/moment because it could result in the date being different.

Ex: `2000-01-01T00:00:00Z == 1999-12-31T23:00:00-01:00` so we need to be careful about how we handle this. It's why just working with MM/dd/yyyy is probably simplest. No time component at all to work with...